subsys/bluetooth/controller/ll_sw/ull_llcp.c

```diff
@@ -71,17 +65,26 @@ static struct mem_pool mem_ctx = { .pool = buffer_mem_ctx };
 /*
  * LLCP Resource Management
  */
-static struct proc_ctx *proc_ctx_acquire(void)
+static struct proc_ctx *proc_ctx_acquire(struct mem_pool *owner)
 {
 	struct proc_ctx *ctx;
 
-	ctx = (struct proc_ctx *)mem_acquire(&mem_ctx.free);
+	ctx = (struct proc_ctx *)mem_acquire(&owner->free);
+
+	if (ctx) {
+		/* Set the owner */
+		ctx->owner = owner;
+	}
+
 	return ctx;
 }
 
 void llcp_proc_ctx_release(struct proc_ctx *ctx)
 {
-	mem_release(ctx, &mem_ctx.free);
+	if (ctx->owner) {
```

Comment:
What if there is no owner? That should not happen but if it happens then released context is lost.